src/main/java/com/rtb/projectmanagementtool/project/ProjectData.java

```diff
@@ -0,0 +1,204 @@
+/**
+ * ProjectData.java - this file implements the Project class, which contains details about a
+ * particular project, including the members and tasks within it.
+ */
+package com.rtb.projectmanagementtool.project;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.HashSet;
+
+enum User {
+  CREATOR,
+  ADMIN,
+  REGULAR;
+}
+
+public class ProjectData {
+  /**
+   * constant variables used to specify the names of each property of the Project entity. Used to
+   * minimize redundancy and also make renaming properties easy if needed
+   */
+  private final String PROPERTY_NAME = "name";
+
+  private final String PROPERTY_DESCRIPTION = "description";
+  private final String PROPERTY_USER_IDS = "userIds";
+  private final String PROPERTY_TASK_IDS = "taskIds";
+
+  private DatastoreService datastore;
+
+  private long id;
+  private String name;
+  private String description;
+  private HashMap<Long, User> userIds;
+  private HashSet<Long> taskIds;
+
+  /**
+   * Class constructor with the minimum requirements for creating a project.
+   *
+   * @param name the name of the project
+   * @param description the description of the project
+   * @param creatorId the id of the user who created the project
+   */
+  public ProjectData(String name, String description, long creatorId) {
+    datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
I would advise separating the database operations from your Data model object.  One way to reason about this is by thinking about the "separation of responsibilities".  You'd want your Data model to be just that, a model/representation for the data.  You'd then want operations done to copes of these done somewhere else (the controller for instance).

Another thought, separating them would allow us later on if we decide to swap to a different flavor of Datastore (or maybe even a different database engine all together), you won't have to update all of your Data models.